Library/Homebrew/utils/curl.rb

```diff
@@ -139,15 +139,15 @@ def curl_output(*args, **options)
     # Check if a URL is protected by CloudFlare (e.g. badlion.net and jaxx.io).
     def url_protected_by_cloudflare?(details)
       [403, 503].include?(details[:status].to_i) &&
-        details[:headers].match?(/^Set-Cookie: __cfduid=/i) &&
-        details[:headers].match?(/^Server: cloudflare/i)
+        details[:headers]&.match?(/^Set-Cookie: __cfduid=/i) &&
```

Comment:
Why does this change result in sometimes-`nil` headers? I'd suggest an early return if `details[:headers]` is `nil` rather than doing a `&` each time.

Reply:
Many of the values in the return hash from `curl_http_content_headers_and_checksum` are `nil` if the server doesn't provide a good response (e.g., no response, timeout, etc.).

The salient change here is that the `headers` value is now `nil` in this scenario instead of an empty string, which is in keeping with the other hash values. The existing logic relied on the value never being `nil`, so it was necessary to update usage to account for this.

That said, I agree an early return would be better in the `url_protected_by_cloudflare?` and `url_protected_by_incapsula?` methods 👍